src/Bicep.LangServer/Handlers/BicepCompletionHandler.cs

```diff
@@ -33,7 +38,15 @@ public override Task<CompletionList> Handle(CompletionParams request, Cancellati
 
             int offset = PositionHelper.GetOffset(compilationContext.LineStarts, request.Position);
             var completionContext = BicepCompletionContext.Create(compilationContext.Compilation.SyntaxTreeGrouping.EntryPoint, offset);
-            var completions = this.completionProvider.GetFilteredCompletions(compilationContext.Compilation, completionContext);
+            var completions = Enumerable.Empty<CompletionItem>();
+            try 
+            {
+                completions = this.completionProvider.GetFilteredCompletions(compilationContext.Compilation, completionContext);
+            } 
+            catch (Exception e)
+            {
+                this.logger.LogError("Error with Completion in file {Uri} with {Context}. Underlying exception is: {Exception}", request.TextDocument.Uri, completionContext, e.ToString());
```

Comment:
What's the behavior if you just let the exception get thrown here? Does the extension crash?

Reply:
I simulated this by throwing an error in the FileResolver purposely.  It simply logs the issue and returns an empty completion. It does not crash